# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1676396028 -3600
# Parent  8901acde30f8767d1996a45954934165730b93d4
Backout bug 1730429 for causing bug 1778091 (lost style).

diff --git a/editor/libeditor/HTMLEditSubActionHandler.cpp b/editor/libeditor/HTMLEditSubActionHandler.cpp
--- a/editor/libeditor/HTMLEditSubActionHandler.cpp
+++ b/editor/libeditor/HTMLEditSubActionHandler.cpp
@@ -655,16 +655,22 @@ nsresult HTMLEditor::OnEndHandlingTopLev
         reapplyCachedStyle = true;
         break;
       default:
         reapplyCachedStyle =
             IsStyleCachePreservingSubAction(GetTopLevelEditSubAction());
         break;
     }
 
+#if 0
+
+Code from https://bugzilla.mozilla.org/show_bug.cgi?id=1730429 - https://hg.mozilla.org/mozilla-central/rev/f895a0b6c0e3
+Causes style loss. It's fixed again in https://bugzilla.mozilla.org/show_bug.cgi?id=1778091 which is
+impossible to backport to 102.
+
     // If the selection is in empty inline HTML elements, we should delete
     // them unless it's inserted intentionally.
     if (mPlaceholderBatch &&
         TopLevelEditSubActionDataRef().mNeedsToCleanUpEmptyElements &&
         SelectionRef().IsCollapsed() && SelectionRef().GetFocusNode()) {
       RefPtr<Element> mostDistantEmptyInlineAncestor = nullptr;
       for (Element* ancestor :
            SelectionRef().GetFocusNode()->InclusiveAncestorsOfType<Element>()) {
@@ -687,16 +693,17 @@ nsresult HTMLEditor::OnEndHandlingTopLev
         if (NS_FAILED(rv)) {
           NS_WARNING(
               "HTMLEditor::DeleteNodeWithTransaction() failed at deleting "
               "empty inline ancestors");
           return rv;
         }
       }
     }
+#endif
 
     // But the cached inline styles should be restored from type-in-state later.
     if (reapplyCachedStyle) {
       DebugOnly<nsresult> rvIgnored = mTypeInState->UpdateSelState(*this);
       NS_WARNING_ASSERTION(NS_SUCCEEDED(rvIgnored),
                            "TypeInState::UpdateSelState() failed, but ignored");
       rvIgnored = ReapplyCachedStyles();
       NS_WARNING_ASSERTION(
